gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiServiceImpl.java

```diff
@@ -688,7 +708,13 @@ public Page<GenericApiEntity> findAll(
             .stream()
             .map(api -> {
                 PrimaryOwnerEntity primaryOwner = primaryOwnerService.getPrimaryOwner(executionContext, api.getId());
-                return genericApiMapper.toGenericApi(api, PrimaryOwnerEntity.builder().id(primaryOwner.getId()).build());
+                var primaryOwnerBuilder = PrimaryOwnerEntity.builder().id(primaryOwner.getId());
+
+                if (expands != null && expands.contains(EXPAND_PRIMARY_OWNER)) {
+                    primaryOwnerBuilder.displayName(primaryOwner.getDisplayName());
+                }
+
+                return genericApiMapper.toGenericApi(api, primaryOwnerBuilder.build());
```

Comment:
If expand does not contain "PrimaryOwner", I would have set PO to null instead of just putting the id.
Nothing in the openapi suggests that the primaryowner object field is required.
And by doing what you did, you are loosing the interest of the 'expand' : reduce the number of calls.

Reply:
Done